src/auth/oauth2/mod.rs

```diff
@@ -96,10 +96,14 @@ impl Inner {
                                     $field: $field.clone(),
                                 )*
                             };
+                            cx.waker().wake_by_ref();
                             break Poll::Pending;
                         }
                     },
-                    Poll::Pending => break Poll::Pending,
+                    Poll::Pending => {
+                        cx.waker().wake_by_ref();
```

Comment:
Just realized, I think only the above wake is necessary. In this case the inner future being polled is responsible for waking.

Reply:
valid point. i assumed the inner future was not doing its job and just added one here. this seem to fix the issue, but i don't have a valid reasoning about this fix though. also, there is a possibility that the issue could be stemming from somewhere else in the library. 

**the issue** -- without this wake, when application-level tasks attempt to fetch the token concurrently, the tasks get hanged after fetching the token and doesn't progress further.

Reply:
Can you try to create a minimal repro?